common/test/acceptance/pages/lms/dashboard.py

```diff
@@ -44,6 +68,60 @@ def _get_course_name(el):
 
         return self.q(css='section.info > hgroup > h3 > a').map(_get_course_name).results
 
+    def check_enrollment_mode(self, course_name, mode='verified'):
+        """Verify enrollment in a given mode.
+
+        Checks if the dashboard contains an enrollment in the provided mode for
+        the provided course name.
+
+        Arguments:
+            course_name (str): The name of the course whose mode should be checked.
+            
+        Keyword Arguments:
+            mode (str): The expected enrollment mode.
+
+        Returns:
+            A Boolean; whether an enrollment in the provided mode for the provided
+                course name was found on the dashboard.
+        """
+        # Filter elements by course name, only returning the relevant course item
+        course_listing = self.q(css=".course").filter(lambda el: course_name in el.text).results
+        
+        if course_listing:
+            # There should only be one course listing for the provided course name
+            banner_classes = course_listing[0].get_attribute('class')
+        else:
+            return False
+
+        return mode in banner_classes
```

Comment:
I've implemented the approach described above, since it makes the dashboard page object more reusable by way of providing additional information to the caller.